selfdrive/ui/tests/test_translations.py

```
@@ -121,6 +123,56 @@ def test_entities_error(self):
         matches = re.findall(r'@(\w+);', cur_translations)
         self.assertEqual(len(matches), 0, f"The string(s) {matches} were found with '@' instead of '&'")
 
+  def test_bad_language(self):
+    """
+    Tests if the given translation files contain any bad language using LDNOOBW.
+    """
+    AVAILABLE_LANGUAGE_CODES = ['ar', 'cs', 'da', 'de', 'en', 'eo', 'es', 'fa', 'fi', 'fil', 'fr', 'hi', 'hu', 'it', 'ja', 'kab', 'ko', 'nl', 'no', 'pl',
+                                'pt', 'ru', 'sv', 'th', 'tlh', 'tr', 'zh']
+    OVERRIDE_WORDS = ['pédale']
+
+    for name, file in self.translation_files.items():
+      match = re.search(r'_([a-zA-Z]{2,3})', file)
+      if not match:
+        print(f"{name} - could not parse language")
+        continue
```

all these print and continues should raise errors, not fail silently